core/src/main/java/org/elasticsearch/common/lucene/all/AllTermQuery.java

```diff
@@ -87,21 +86,18 @@ public Query rewrite(IndexReader reader) throws IOException {
         if (rewritten != this) {
             return rewritten;
         }
-        boolean fieldExists = false;
         boolean hasPayloads = false;
         for (LeafReaderContext context : reader.leaves()) {
             final Terms terms = context.reader().terms(term.field());
             if (terms != null) {
-                fieldExists = true;
                 if (terms.hasPayloads()) {
                     hasPayloads = true;
                     break;
                 }
             }
         }
-        if (fieldExists == false) {
-            return new MatchNoDocsQuery();
-        }
+        // if the terms does not exist we could return a MatchNoDocsQuery but this would break the unified highlighter
+        // which rewrites query with an empty reader.
```

Comment:
Is this just going to make queries again `_all` a little less efficient what `_all` is disabled? If so I think that is ok.